src/service/tempMailApi.js

```diff
@@ -6,11 +6,24 @@ class TemporalMailService {
   constructor() {
     this.BASE_API_URL = 'https://api.mail.tm';
     this.httpClient = require('axios');
+    this.emailDomain = '';
   }
+
+  async _getDomains() {
+    try {
+      const response = await this.httpClient.get(`${this.BASE_API_URL}/domains`);
+      const {domain} = response.data['hydra:member'];
+      this.emailDomain = domain;
+    } catch (err) {
+      throw new Error('Cant get domains for emails');
+    }
+  }
+
   async createMail({mail, password}) {
     try {
+      await this._getDomains();
```

Comment:
If this is something that you're calling in a global context, why you call everytime that you want to create a email?